src/main/java/org/opentest4j/fixtures/FixtureContext.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2015-2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.opentest4j.fixtures;
+
+/**
+ * Provides methods to register {@link TearDown} and {@link TestPostcondition} instances.
+ *
+ * @author Kevin Cooney
+ * @since 1.1
+ */
+public abstract class FixtureContext {
+
+	/**
+	 * Adds a {@link TearDown} to run after the test completes.
+	 */
+	public abstract void addTearDown(TearDown tearDown);
```

Comment:
What would `addSetUp()` to that would be different than what happens in the current fixture initialization? Are you proposing adding more methods to `TestFixture`?

The reason why I don't have `addSetUp()` is because fixture initialization almost always requires tear-down work (if the fixture delegates to other fixtures or classes, it might require registering multiple tear-downs). That's why initialization of a test fixture happens in `TestFixture.initialize(FixtureContext)`.